### PR TITLE
Allows configuring custom field tooltip for Zapp UI

### PR DESCRIPTION
**Pull request by vinagrito**
@ranmeirman 
Here we ask from the plugin configuration creator to add a text that will be used in Zapp UI in order to make it more clear when navigating through custom added fields

Update

The manifest will now have:

```JSON
...
"custom_configuration_fields": [
    {
      "type": "text",
      "key": "welcome",
      "tooltip_text": "This should appear when launching the login screen",    // Added
      "default": "Hello!, world"
    },
   {
      "type": "dropdown",
      "key": "data_source",
      "tooltip_text": "This will point to the source to be used for the data origin", //Added
      "multiple": false,
      "options": [
        "appli2",
        "3rdparty"
      ],
      "default": "appli2"
    }
  ]
...
```

### CODE REVIEW

File: lib/question.rb [hunk 1]
```diff
@@ -16,6 +16,15 @@ def ask_non_whitespaces(question, field_name)
   end
 
   def ask_base(question)
-    ask("[?] #{question}")
+    ask("[?] #{question}") do |q|
+      yield(q) if block_given?
+    end
+  end
+
+  def required_with_min_length(question, field_name, min_length = 10)
```
Comment (ranmeirman) [inline lib/question.rb]: shouldn't it have maximum length as well? unless you are planning to cut it in the UI

Comment (vinagrito) [inline lib/question.rb]: I don't know. I guess this goes along with the/a product definition. The spec doesn't include this. What should it be?

Comment (ranmeirman) [inline lib/question.rb]: it is kind of defining where you wanna do the validation... I think the best way is on the Zappifest but I would not go to court with you :)

**Comment by vinagrito** [comment]
No longer setting the type in the manifest. Zappifest sets the tooltip content for Zapp to display

**Pull request merged**